### PR TITLE
Upgrade operator-sdk from v0.12.0 to v0.16.0

### PR DESCRIPTION
**Pull request by dustman9000**
Bring operator-sdk up to v0.16.0 for further standardizations using the [upgrade guide](https://sdk.operatorframework.io/docs/upgrading-sdk-version/version-upgrade-guide/#v016x) as reference.

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -142,32 +182,24 @@ func main() {
...
-	if err != nil {
-		return err
-	}
-	// To generate metrics in other namespaces, add the values below.
-	ns := []string{operatorNs}
+
+	// The metrics will be generated from the namespaces which are returned here.
+	// NOTE that passing nil or an empty list of namespaces in GenerateAndServeCRMetrics will result in an error.
+	// Due to https://github.com/operator-framework/operator-sdk/issues/2707
+	ns := []string{""}
```
Comment (2uasimojo) [inline cmd/manager/main.go:201]: Is this a regression? Do we need to include `operatorNs`?

Comment (dustman9000) [reply cmd/manager/main.go]: Good catch. This was in preparation for making the operator cluster-scoped as there is an [issue](https://github.com/operator-framework/operator-sdk/issues/2707) with metrics and cluster-scoped operators. I've changed it back to the recommended code and will fix this if/when we decide to make the operator cluster-scoped.

File: cmd/manager/main.go [hunk 2]
```diff
@@ -142,32 +182,26 @@ func main() {
...
 	filteredGVK, err := k8sutil.GetGVKsFromAddToScheme(apis.AddToScheme)
 	if err != nil {
 		return err
 	}
-	// Get the namespace the operator is currently deployed in.
-	operatorNs, err := k8sutil.GetOperatorNamespace()
+
+	// The metrics will be generated from the namespaces which are returned here.
+	// NOTE that passing nil or an empty list of namespaces in GenerateAndServeCRMetrics will result in an error.
+	ns, err := kubemetrics.GetNamespacesForMetrics(operatorNs)
```
Comment (2uasimojo) [inline cmd/manager/main.go:200]: This should work, but is redundant with the [`operatorNs` param](https://github.com/openshift/must-gather-operator/pull/9/files#diff-9e63f7b0791edcb4a53c944552b88970R147) that now gets [passed into this method](https://github.com/openshift/must-gather-operator/pull/9/files#diff-9e63f7b0791edcb4a53c944552b88970R155).

**Review by 2uasimojo — COMMENTED** [review]
Nonblocking observation within.

Nonmerging lgtm (@csheremeta to approve for real).

**Pull request merged**